src/QsCompiler/Tests.Compiler/TestCases/TypeChecking.qs

```diff
@@ -843,6 +843,18 @@ namespace Microsoft.Quantum.Testing.TypeChecking {
         return () -> 0;
     }
 
```

Comment:
Is there a test existing that has multiple operation sub-expressions in the lambda with differing characteristics to test that the lambda's characteristics are the intersection of all the operation sub-expressions?

Reply:
No, I don't think there was test coverage of that, thanks for pointing that out. I added a few tests.

Reply:
Looks good, thanks!